src/fastapi_poe/base.py

```diff
@@ -307,6 +320,25 @@ async def handle_query(
                         linkify=event.linkify,
                         suggested_replies=event.suggested_replies,
                     )
+                elif isinstance(event, AttachFileResponse):
+                    upload_task = asyncio.create_task(
+                        request.post_message_attachment(
+                            file_data=event.file_data,
+                            filename=event.filename,
+                            content_type=event.content_type,
+                            is_inline=event.is_inline,
+                        )
+                    )
+                    if event.is_inline:
```

Comment:
What happens if this is not inline?

Reply:
Then we don't need to wait for the upload task to finish to get the inline ref and we don't need to add any markdown text to the response so we can skip the code in this block. The way this is implemented, the bot can continue to yield back more text while the non-inline file upload is still in flight.

One downside with this approach is that the response won't error out if the attachment upload fails. If we wanted to do that, we would have to wait on the `upload_task`.

Reply:
I kinda feel like users would expect this to work synchronously (including raising errors if the request fails)